packages/core/src/plugins/default/markbind-plugin-plantuml.ts

```diff
@@ -28,16 +51,23 @@ let graphvizCheckCompleted = false;
  * @param content puml dsl used to generate the puml diagram
  */
 function generateDiagram(imageOutputPath: string, content: string) {
+  const hashKey = cryptoJS.MD5(imageOutputPath + content).toString();
+
   // Avoid generating twice
-  if (processedDiagrams.has(imageOutputPath)) { return; }
-  processedDiagrams.add(imageOutputPath);
+  if (processedDiagrams.has(hashKey)) {
```

Comment:
should we check the status of `isAlive` here? Is there different treatment for either value?

Reply:
Don't need to check; we just need to set it to isFresh to ensure that the diagram is still being used by the site.